apps/babanews/pages/a/tag/[tag].tsx

```diff
@@ -60,6 +61,10 @@ export default function ArticleListByTag({tagId}: ArticleListByTagProps) {
     return 1
   }, [data?.articles.totalCount])
 
+  const canonicalUrl = useMemo(() => {
```

Comment:
No need to useMemo for this as not enough CPU cycles are saved (and possibly even more used) + extra memory.